File: app/src/test/java/com/mgaetan89/showsrage/fragment/ScheduleFragment_GetSectionNameTest.java
package com.mgaetan89.showsrage.fragment;

import com.mgaetan89.showsrage.R;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(Parameterized.class)
public class ScheduleFragment_GetSectionNameTest {
	@Parameterized.Parameter(0)
	public String sectionId;

	@Parameterized.Parameter(1)
	public int sectionName;

	@Test
	public void getSectionName() {
		assertThat(ScheduleFragment.getSectionName(this.sectionId)).isEqualTo(this.sectionName);
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		return Arrays.asList(new Object[][]{
				{null, 0},
				{"", 0},
				{"later", R.string.later},
				{"Later", R.string.later},
				{"missed", R.string.missed},
				{"Missed", R.string.missed},
				{"soon", R.string.soon},
				{"Soon", R.string.soon},
				{"today", R.string.today},
				{"Today", R.string.today},
				{"statusid", 0},
				{"StatusId", 0},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/fragment/ScheduleFragment_GetSectionNameTest.kt
package com.mgaetan89.showsrage.fragment

import com.mgaetan89.showsrage.R
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class ScheduleFragment_GetSectionNameTest(val sectionId: String?, val sectionName: Int) {
	@Test
	fun getSectionName() {
		assertThat(ScheduleFragment.getSectionName(this.sectionId)).isEqualTo(this.sectionName)
	}

	companion object {
		@JvmStatic
		@Parameterized.Parameters
		fun data(): Collection<Array<Any?>> {
			return listOf(
					arrayOf<Any?>(null, 0),
					arrayOf<Any?>("", 0),
					arrayOf<Any?>("later", R.string.later),
					arrayOf<Any?>("Later", R.string.later),
					arrayOf<Any?>("missed", R.string.missed),
					arrayOf<Any?>("Missed", R.string.missed),
					arrayOf<Any?>("soon", R.string.soon),
					arrayOf<Any?>("Soon", R.string.soon),
					arrayOf<Any?>("today", R.string.today),
					arrayOf<Any?>("Today", R.string.today),
					arrayOf<Any?>("statusid", 0),
					arrayOf<Any?>("StatusId", 0)
			)
		}
	}
}
